### PR TITLE
Refactor input to neural network

### PR DESCRIPTION
**Pull request by gogo2245**
This PR allows us to use input data in vector format

### CODE REVIEW

File: prepare_dataset.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import random
+import torch
+import os
+
+curr_path = os.path.abspath(os.getcwd())
+dir_path = os.path.join(curr_path, 'data')
+_, _, file_names = next(os.walk(dir_path))
+data = []
+labels = {}
+counter = 0
```
Comment (martinPasen) [inline prepare_dataset.py:10]: what is the counter for ?

File: prepare_dataset.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        rand_label = random.choice(list(labels.keys()))
+        rand = labels[rand_label]
+        bytes, _, cur = data[rand]
+        input_data += bytes[cur * 300:(cur + 1) * 300]
+        target += [rand] * 300
+        data[rand][2] += 1
+        counter -= 1
+        if data[rand][1] - 1 == data[rand][2]:
+            labels.pop(rand_label)
+    if len(input_data) == 1500:
```
Comment (martinPasen) [inline prepare_dataset.py:35]: we are using 1500 on too many places ... probably it is time to refactor it to some constant/variable

File: transforms.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+class To256(object):
+    def __call__(self, input, device):
+        tn = torch.zeros((1500, 256), device=device, dtype=torch.bool)
+        for i, j in enumerate(input):
+            tn[i][j] = True
+        return tn
+
+    def __repr__(self):
+        return self.__class__.__name__ + '()'
```
Comment (martinPasen) [inline transforms.py:12]: add eof

**Review by martinPasen — CHANGES_REQUESTED** [review]

File: prepare_dataset.py (hunk 1 above)
Comment (gogo2245) [reply prepare_dataset.py:10]: Sorry this whole file is not needed I used it to experiment and then found out I did not need it

**Comment by gogo2245** [comment]
@martinPasen sorry for the inconvenience with this `prepare_dataset.py` file. Now it's as it should be

**Review by martinPasen — APPROVED** [review]

**Pull request merged**